x/dex/types/keys.go

```diff
@@ -55,10 +55,10 @@ func PricePrefix(contractAddr string) []byte {
 	return append(KeyPrefix(PriceKey), KeyPrefix(contractAddr)...)
 }
 
-func SettlementEntryPrefix(contractAddr string, blockHeight uint64) []byte {
+func SettlementEntryPrefix(contractAddr string, priceDenom string, assetDenom string) []byte {
 	return append(
 		append(KeyPrefix(SettlementEntryKey), KeyPrefix(contractAddr)...),
-		GetKeyForHeight(blockHeight)...,
+		PairPrefix(priceDenom, assetDenom)...,
```

Comment:
we will have a catch-all migration for all recent `dex` changes where the migration handler nukes everything under the `dex` namespace, since we don't have any real dex use cases yet